src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -1336,7 +1338,16 @@ public Runnable getEditsRunnable(VirtualFile file, int version,
                         document.insertString(start, text);
                     }
                 }
-                FileDocumentManager.getInstance().saveDocument(document);
+                try {
+                    FileDocumentManager.getInstance().saveDocument(document);
+                    PsiDocumentManager.getInstance(project).commitDocument(document);
+                    DidChangeTextDocumentParams didChange = new DidChangeTextDocumentParams();
+                    didChange.setTextDocument(new VersionedTextDocumentIdentifier(FileUtils.sanitizeURI(file.getUrl()), version));
+                    didChange.setContentChanges(Collections.singletonList(new TextDocumentContentChangeEvent(document.getText())));
+                    wrapper.getRequestManager().didChange(didChange);
```

Comment:
Why are we sending a `didchange` request in here?

Reply:
If an edit is applied to a document that has not yet been opened in the editor, the Language Server does not recognize that the content has changed. This results in error diagnostics being displayed in the document where the renaming was initiated. To prevent this, a 'didChange' request is sent along with the renamed files.

Reply:
@ThemiraChathumina got the reason. sill I'm not sure if its fine to send `didChange` events for the files which are not open in the editor (which means that the `didOpen` event is not triggered for them).